### PR TITLE
Test Qos with ovs bridge

### PR DESCRIPTION
**Pull request by yalzhang**
Test the inbound and outbound of bridge type interface with ovs bridge.

Signed-off-by: Yalan <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
$ avocado run --vt-type libvirt iface_ovs
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 2f3259075a9d69ac6f138e70e5cfd30e17afa9fa
JOB LOG    : /root/avocado/job-results/job-2021-12-07T21.32-2f32590/job.log
 (1/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.default_test: PASS (9.53 s)
 (2/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_portgroup.portgroup_42: PASS (11.79 s)
 (3/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_portgroup.portgroup_4: PASS (9.60 s)
 (4/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_iface: PASS (9.22 s)
 (5/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_qos.start: PASS (9.38 s)
 (6/6) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_qos.hotplug_iface: PASS (9.57 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 62.39 s

depend on https://github.com/avocado-framework/avocado-vt/pull/3292

**Comment by yalzhang** [comment]
one more case and all passed locally
$ avocado run --vt-type libvirt iface_ovs
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : fe367598e740ed4953a170f481a4717fbde0bfbb
JOB LOG    : /root/avocado/job-results/job-2021-12-08T04.39-fe36759/job.log
 (1/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.default_test: PASS (9.73 s)
 (2/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_portgroup.portgroup_42: PASS (9.66 s)
 (3/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_portgroup.portgroup_4: PASS (10.27 s)
 (4/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_iface: PASS (9.24 s)
 (5/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_qos.start: PASS (10.92 s)
 (6/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_qos.hotplug_iface: PASS (10.84 s)
 (7/7) type_specific.io-github-autotest-libvirt.virtual_network.iface_ovs.net_ovs.ovs_qos.domiftune_add: PASS (9.48 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 74.70 s

**Comment by yalzhang** [comment]
@Yingshun Please help to review, Thank you!

File: libvirt/tests/cfg/virtual_network/iface_ovs.cfg [hunk 1]
```diff
@@ -32,3 +32,21 @@
...
+                    iface_bandwidth_inbound = "{'average':'2000','peak':'3000', 'burst':'1000'}"
+                    iface_bandwidth_outbound = "{'average':'5000','peak':'3000','burst':'1000'}"
+                    test_qos = "yes"
+                    variants:
+                        - start:
+                        - hotplug_iface:
+                            hotplug = "yes"
+                        - domiftune_add:
+                            hotplug = 'yes'
+                            live_add_qos = 'yes'
```
Comment (Yingshun) [inline libvirt/tests/cfg/virtual_network/iface_ovs.cfg:51]: Do you mind explaining the purpose of the test? I can't understand its real meaning from the variant names. Sorry, I'm not familiar with this feature😅  ...

File: libvirt/tests/src/virtual_network/iface_ovs.py [hunk 2]
```diff
@@ -140,17 +126,46 @@ def check_ovs_port(ifname, brname):
...
 
         try:
             # Start the VM.
             vm.start()
             if start_error:
                 test.fail("VM started unexpectedly")
-
+            if hotplug:
+                res = virsh.attach_device(vm_name, iface_attach_xml, debug=True)
+                libvirt.check_exit_status(res)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_ovs.py:150]: If no expect_error param，I think no need to use check_exist_stauts to check the return, but just give ignore_status=False in the virsh cmd

File: libvirt/tests/src/virtual_network/iface_ovs.py [hunk 3]
```diff
@@ -140,17 +126,46 @@ def check_ovs_port(ifname, brname):
...
             iface_name = libvirt.get_ifname_host(vm_name, iface_mac)
             if test_ovs_port:
                 check_ovs_port(iface_name, bridge_name)
+            if live_add_qos:
+                inbound_opt = ",".join(re.findall(r'[0-9]+', iface_inbound))
+                outbount_opt = ",".join(re.findall(r'[0-9]+', iface_outbound))
+                res = virsh.domiftune(vm_name, iface_name, inbound=inbound_opt,
+                                      outbound=outbount_opt, debug=True)
+                libvirt.check_exit_status(res)
```
Comment (kylazhang) [inline libvirt/tests/src/virtual_network/iface_ovs.py:159]: As above

File: libvirt/tests/cfg/virtual_network/iface_ovs.cfg (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/cfg/virtual_network/iface_ovs.cfg:51]: 1. start vm with bridge type interface connected to ovs bridge, and the interface set with bandwidth setting.
check point: the guest can start successfully; and qos test by tc;
2. start vm without any interface, then hotplug interface  connected to ovs bridge, and the interface set with bandwidth setting;
check point: the hotplug succeed; and qos test by tc;
3. start vm without any interface, then hotplug interface  connected to ovs bridge without any qos setting, then set bandwidth by domiftune;
checkpoint: the bandwidth setting set successfully; and qos test by tc;
Maybe the scenarios need some update, feel free to add your comment, Thank you!

**Comment by Yingshun** [comment]
@yalzhang rebase please...

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by kylazhang — APPROVED** [review]

**Pull request merged**